packages/changeset-form/addon/components/changeset-form/index.ts

```diff
@@ -18,49 +14,32 @@ interface ChangesetFormArgs {
 }
 
 export default class ChangesetForm extends Component<ChangesetFormArgs> {
-  changeset!: BufferedChangeset;
-
   @tracked hasSubmitted = false;
 
   constructor(owner: unknown, args: ChangesetFormArgs) {
     super(owner, args);
-
-    if (typeof this.args.changeset === 'undefined') {
-      assert(
-        '@model must be defined on <ChangesetForm> component if you do not provide a @changeset argument',
-        typeof this.args.model !== 'undefined'
-      );
-
-      if (typeof this.args.validations !== 'undefined') {
-        this.changeset = Changeset(
-          this.args.model || {},
-          lookupValidator(this.args.validations || {}),
-          this.args.validations
-        );
-      } else {
-        this.changeset = Changeset(this.args.model || {});
-      }
-    } else {
-      this.changeset = this.args.changeset;
-    }
+    assert('Must passs down a changeset', this.args.changeset);
   }
 
   @action
-  async handleSubmit(event: Event): Promise<void> {
+  async handleSubmit(
+    changeset: BufferedChangeset,
```

Comment:
Any reason to not use the changeset from the args?

Reply:
I had issues between tests, like if the a new changeset was being created from test to test, thus leading to incorrect validatios, I don't think this would happen in a real situation? also, I do think we could also yield these functions to let the client use them for ergonomics...